pkg/controller/scylladbdatacenter/resource.go

```diff
@@ -653,12 +653,32 @@ exec /mnt/shared/scylla-operator sidecar \
 --service-name=$(SERVICE_NAME) \
 --cpu-count=$(CPU_COUNT) \
 --loglevel=2 \
-` + func() string {
-										if len(sdc.Spec.ScyllaDB.ExternalSeeds) > 0 {
-											return fmt.Sprintf("--external-seeds=%s", strings.Join(sdc.Spec.ScyllaDB.ExternalSeeds, ","))
-										}
-										return ""
-									}(),
+` +
+										func() string {
+											var optionalArgs []string
+
+											if len(sdc.Spec.ScyllaDB.ExternalSeeds) > 0 {
+												optionalArgs = append(optionalArgs, fmt.Sprintf("--external-seeds=%s", strings.Join(sdc.Spec.ScyllaDB.ExternalSeeds, ",")))
+											}
+
+											return strings.Join(optionalArgs, ` \`)
+										}() +
+										func() string {
+											var positionalArgs []string
+
+											if len(sdc.Spec.ScyllaDB.AdditionalScyllaDBArguments) > 0 {
+												positionalArgs = append(positionalArgs, sdc.Spec.ScyllaDB.AdditionalScyllaDBArguments...)
+											}
+
+											if sdc.Spec.ScyllaDB.EnableDeveloperMode != nil && *sdc.Spec.ScyllaDB.EnableDeveloperMode {
+												positionalArgs = append(positionalArgs, "--developer-mode=1")
+											}
+
+											if len(positionalArgs) > 0 {
+												return fmt.Sprintf(" -- %s \n", strings.Join(positionalArgs, " "))
```

Comment:
Shouldn't we escape each arg with `%q` so it propagates correctly?

Security wise, given that the args are supplied by the user we should prevent injecting bash snippets as well (subshell, reading env vars).

Reply:
to avoid escaping, maybe supplying the args to the bash itself and then propagating them with `-- "$@"` might work
```bash
bash -c 'echo "$@"' -- foo bar
```

Reply:
done, tested with injected subshell printing secret and it doesn't work.